lib/services/s3.js

```diff
@@ -411,7 +409,10 @@ AWS.util.update(AWS.S3.prototype, {
     // This chunk of code will set the location constraint param based
     // on the region (when possible), but it will not override a passed-in
     // location constraint.
-    if (!params) params = {};
+    if (typeof params === 'function') {
+      callback = params;
+      params = {};
+    }
```

Comment:
It might be useful to add one more check after the if block to set params equal to itself or and empty object to protect us in the unlikely event someone passes in a value like `null`. Imagine this would be a rare edge case but we do this already for other operations.